### ISSUE TITLE
Running the code

### ISSUE THREAD

**Issue by YasminAMassoud** [garethjns/Kaggle-EEG#13]
Hello 

I am trying to run the code but i currently have 2 issues :

1)I have only original Kaggla data, so not sure how to modify the code as i dont have the new test set 
2) trainedModelsCompactTest.mat i dont have this file so it give me error while running the code 

Thanks

**Comment by garethjns** [comment]
Hi Yasmin,

I'm afraid my memory of the Kaggle data is a bit hazy. I think a second test set was provided along with a list of leaks in the old test set that could be moved into the training set. This is what the copyTestLeakToTrain.m script does. I think you might be able to skip this step, and manually create the directory structure (as in the readme) but just using the original data. 

The trainedModelsCompactTest.mat should be produced by train.m if you can get it to run with the data you have.

**Comment by YasminAMassoud** [comment]
@garethjns  Thanks for your input in this issue ,I have solved the trainedmodelscompacttest.mat now. 
*For issue 2 )  I tried this is then have and error 
Unable to read file singles1.mat , which is i believe is output from copytestleaktotrain.mat , It will be helpful if you give me advice on how to obtain this file without using copytestleaktrain.mat 

*Also the New folder now will have files named pat1test_1_0.mat for example they wont be _new will this create a problem ?

**Comment by garethjns** [comment]
If I remember correctly, the singles_n.mat files have a list of files for each subject where the file is an independent 10 minutes of recording, rather than a 10 min segment of a consecutive 60 min recording (spread over 6 files). I think all of these will have originally been in the test set (which were all supposed to be independent 10 minute segments), before the leak was discovered. This would make sense if the singles.mat files are created by copyTestLeakToTrain.m.

I think if you just have the original set it might be worth trying to modify the method that loads the training files. This has logic to load the group 10 min segments of the 60 min groups, and logic to load the single files, and mark them as single so they aren't concatenated together.

This method is here:
 https://github.com/garethjns/Kaggle-EEG/blob/c8883b1b1371b89781b2f82f412559ddbca5f362/%40featuresObject/featuresObject.m#L441 It looks like it loads all the '*.mat' files from the directories it finds in paths.dataDir, so the _new in the files names shouldn't matter as long as paths.DataDir is pointing to the correct place.

I've found copies of the singles.mat files, but I don't have access to MATLAB so can't check their contents at the moment. I've pushed them to the root of the reop on this branch https://github.com/garethjns/Kaggle-EEG/tree/singles_files, but you may not need them. Actually, another be worth trying is to use them but leave them empty (as there are no single files in your training set). That might mean you wouldn't need to modify the method that loads the files.

**Comment by YasminAMassoud** [comment]
-ok so now there are two options, i understand the second one ,which is leaving the file empty however this gives an error meaning that safe4 which is used to save files in singles.mat is non-existant , what i understand from _copytestleaktotrain_ is that _safe4_ is filenames of _testfile_ names in _train&test_data_labels_safe.csv_ , i dont know what will be their use in featureobject, **as my test file names looks like this :** 
![image](https://user-images.githubusercontent.com/55138409/78292177-ecc8a180-7526-11ea-95e4-3999025d0402.png)]
**However test files in _train&test_data_labels_safe.csv looks like this :** 
![image](https://user-images.githubusercontent.com/55138409/78292324-24cfe480-7527-11ea-9bfe-0698e979d560.png) , so this doesnt work due to safe4 files and i am not sure how to modify this. 
So I dont have the test files in train&test_data_labels_safe.csv  , it will be helpful if you can give insights on this as i already understand copytestleaktotrain code 
I just need to understand the exact forms of the all outputs from copytestleaktotrain to make it work fine for the rest again 
To recap : 
*I made my folders with same as folderstructure3.png ,however they all have the same file (original and New Folders) 
*data.Dir and doldTest points to my original folders 
*data.New and dTrain points to my new folders 
*safe4 generations seems redundant or incompatible with rest of code as test files in train&test_data_labels_safe.csv arent compatible with my names in test folders ! 

For the the fist method of editing the genFileListSomeSingles(subs, str, paths) i will take a look at it .

Thanks alot for your time :)

**Comment by YasminAMassoud** [comment]
@garethjns 
Thanks for you reply, i have been trying to re-implement that part, i created 3 singles.mat file containing single files which have different order in new data on ecosystem. 

As for the genFileSomeSingles ,I am having trouble running and understanding the function of this part of the code whats the usage of repmat and reshape here ? As far as i know this is how to know segment details from ecosystem website 

> It is possible to determine which 10 minute segments belong to specific 1 hour blocks for the training data by grouping every 6 blocks based on the index J in 'PatITrain_J_K' (i.e. block index = ceil(J/6) ). Determining the position of a segment within a block can be done using the modulus operator (i.e. segment position = ((J-1) % 6)+1 )
>

`  fileList.Segment(find(exSinglesIdx)) = ...
                    reshape(repmat((1:nFilesSubEx/6)', 1, 6)',nFilesSubEx,1); %#ok<FNDSB>
                fileList.nInSegment(find(exSinglesIdx)) = repmat((1:6)', ...
                    nFilesSubEx/6, 1); %#ok<FNDSB>
                
                % Now label the remaing as single segments
                fileList.Segment(find(~exSinglesIdx)) = ...
                    max(fileList.Segment)+1:...
                    (numel(singles.safe4)+ max(fileList.Segment)); %#ok<FNDSB>
                fileList.nInSegment(find(~exSinglesIdx)) = 1; %#ok<FNDSB>`

I would be grateful if you explain this out

**Comment by YasminAMassoud** [comment]
Thanks to @garethjns , i managed to do it by editing copytestleaktotrain to have singles.mat file according to new data description on ecosystem website .
